helix-rest/src/main/java/org/apache/helix/rest/server/ServerContext.java

```diff
@@ -111,20 +114,9 @@ public RealmAwareZkClient getRealmAwareZkClient() {
       synchronized (this) {
         if (_zkClient == null) {
           // If the multi ZK config is enabled, use FederatedZkClient on multi-realm mode
-          if (Boolean.parseBoolean(System.getProperty(SystemPropertyKeys.MULTI_ZK_ENABLED))) {
-            LOG.info("ServerContext: initializing FederatedZkClient with routing ZK at {}!",
-                _zkAddr);
+          if (_isMultiZkEnabled || Boolean
+              .parseBoolean(System.getProperty(SystemPropertyKeys.MULTI_ZK_ENABLED))) {
```

Comment:
Aren't we moving away from system properties? 
If we allow both it could be confusing, unless we specifically document for it.

Reply:
> Aren't we moving away from system properties?
Why?

> If we allow both it could be confusing, unless we specifically document for it.
This is for convenience. Users that have many namespaces can simply set the system property. Otherwise for more fine-grained control, they can set it at the namespace granularity.

Reply:
Then the system config should have a default value = "false". When user wants to enable fine-grained control, they shouldn't be manually setting `MULTI_ZK_ENABLED = 'false'` and then setting the per-namespace values; that will only be confusing.

Reply:
It is false if it's not set by default.